packages/tx-construction/src/tx-builder/TxBuilder.ts

```diff
@@ -117,17 +118,16 @@ export class GenericTxBuilder implements TxBuilder {
   }
 
   addOutput(txOut: OutputBuilderTxOut): TxBuilder {
-    this.partialTxBody = { ...this.partialTxBody, outputs: [...(this.partialTxBody.outputs || []), txOut] };
-
-    if (txOut.handle) {
-      this.#handles = [...this.#handles, txOut.handle];
+    if (txOut.handleResolution) {
+      this.#handleResolutions = [...this.#handleResolutions, txOut.handleResolution];
     }
-
+    const txOutNoHandle = omit(txOut, 'handle');
```

Comment:
This was updated in original commit
```suggestion
    const txOutNoHandle = omit(txOut, 'handleResolution');
```